### ISSUE TITLE
Bug in categorical time series forecasting

### ISSUE THREAD

**Issue by paxcema** [mindsdb/lightwood#851]
As reported [here](https://github.com/mindsdb/mindsdb/issues/2026), a time series model for categorical targets may fail with the following error:

```
error in apply predictor step: too many indices for array: array is 1-dimensional, but 2 were indexed
Traceback (most recent call last):
  File "./mindsdb/api/mysql/mysql_proxy/classes/sql_query.py", line 739, in execute_step
    data = dn.select(
  File "./mindsdb/api/mysql/mysql_proxy/datahub/datanodes/mindsdb_datanode.py", line 289, in select
    pred_dicts, explanations = self.model_interface.predict(table, where_data, 'dict&explain')
  File "./mindsdb/utilities/with_kwargs_wrapper.py", line 43, in wrapper
    return method(*args, **kwargs)
  File "./mindsdb/interfaces/model/model_interface.py", line 16, in predict
    return self.controller.predict(*args, **kwargs)
  File "./mindsdb/utilities/functions.py", line 56, in wrapper
    return func(*args, **kwargs)
  File "./mindsdb/interfaces/model/model_controller.py", line 301, in predict
    predictions = self.predictor_cache[name]['predictor'].predict(df)
  File "/home/maxs/dev/mdb/venv38/sources/lightwood/lightwood/helpers/log.py", line 30, in wrap
    result = f(predictor, *args, **kw)
  File "/tmp/c51a2b709bc1cc1770171fd678273be16b86f574192f49f516457940293826993.py", line 522, in predict
    df = self.ensemble(encoded_ds, args=self.pred_args)
  File "/home/maxs/dev/mdb/venv38/sources/lightwood/lightwood/ensemble/best_of.py", line 68, in __call__
    raise(e)
  File "/home/maxs/dev/mdb/venv38/sources/lightwood/lightwood/ensemble/best_of.py", line 65, in __call__
    return mixer(ds, args=args)
  File "/home/maxs/dev/mdb/venv38/sources/lightwood/lightwood/mixer/neural.py", line 359, in __call__
    ydf[f'__mdb_proba_{label}'] = raw_predictions[:, idx]
IndexError: too many indices for array: array is 1-dimensional, but 2 were indexed
```

**Comment by fishnux** [comment]
I believe the same will happen with:
https://docs.mindsdb.com/sql/tutorials/crop-prediction/

I'm following that tutorial (although with a different dataset, not crop prediction) and I'm running into the same issue

**Comment by paxcema** [comment]
Closed via #855
